app/models/agents/attribute_difference_agent.rb

```diff
@@ -0,0 +1,104 @@
+require 'bigdecimal'
+
+module Agents
+  class AttributeDifferenceAgent < Agent
+    cannot_be_scheduled!
+
+    description <<-MD
+      The Attribute Difference Agent receives events and emits a new event with
+      the difference or change of a specific attribute in comparison to the previous
+      event received.
+
+      `attribute` specifies the name of the attribute to be used from the event.
+
+      `method` specifies if it should be...
+
+      * `percentage_change` eg. Previous value was 160, new value is 116. Percentage change is `-27.5`
+      * `decimal_difference` eg. Previous value was 5.5, new value is 15.2. Difference is `9.7`
+      * `integer_difference` eg. Previous value was 50, new value is 40. Difference is `-10`
+
+      `expected_update_period_in_days` is used to determine if the Agent is working.
+
+      The resulting event will be a copy of the received event with the difference
+      or change added as an extra attribute. If you use the `percentage_change` the
+      attribute will be formatted as such `{{attribute}}_change`, otherwise it will
+      be `{{attribute}}_diff`.
+    MD
+
+    event_description <<-MD
+      This will change based on the source event.
+    MD
+
+    def default_options
+      {
+        'attribute' => 'rate',
+        'method' => 'integer_difference',
+        'expected_update_period_in_days' => 1
+      }
+    end
+
+    def validate_options
+      unless options['attribute'].present? && options['method'].present? && options['expected_update_period_in_days'].present?
+        errors.add(:base, 'The attribute, method and expected_update_period_in_days fields are all required.')
+      end
+    end
+
+    def working?
+      event_created_within?(interpolated['expected_update_period_in_days']) && !recent_error_logs?
+    end
+
+    def receive(incoming_events)
+      incoming_events.each do |event|
+        handle(interpolated(event), event)
+      end
+    end
+
+    private
+
+    def handle(opts, event)
+      attribute = opts['attribute']
+      attribute_value = event.payload[attribute].nil? ? 0 : event.payload[attribute]
```

Comment:
Oh, I just noticed you are already calling `interpolated(event)` in `receive`, so using liquid to modify the attribute based on the incoming event would already work, but one could not access values nested in the received event object.
You could change the `attribute` option key to `value` and directly use that interpolated result, this would allow the user to set it to `{{data.value}}`, `interpolated({data: {value: 6}})` would then return `{value: "6"}`.